### ISSUE TITLE
Two Chains in One File

### ISSUE THREAD

**Issue by kevroy314** [msoedov/langcorn#21]
Seems like this issue is present if you try to define two chains in one file:

https://github.com/tiangolo/fastapi/discussions/7455

Simple workaround is obviously "don't do that", but often it's useful to define two chains with related purposes in the same file. 

I believe the real fix is to ensure that the derived class name is unique not just to the module but to the chain. The endpoint prefix would also need to be unique. So on this line:

https://github.com/msoedov/langcorn/blob/e36e83147d43dcb476a5c984f081330bd56a8d6b/langcorn/server/api.py#L128

Make it read something like:

```
endpoint_prefix = lang_app.replace(":", '.')
```

That way the function name stays with the info all the way through. I haven't tested this fix, but I think it may work if someone has time to test it.

**Comment by msoedov** [comment]
Hi, @kevroy314, thx for reporting the issue. Can you share your code that reproduces this issue? By the way, thx again for your PR!

**Comment by kevroy314** [comment]
Sure - so if you slightly modify the readme example so that you have two files, `main.py` and `langcorn_two_chain.py` where their contents are:

`main.py`
```
from fastapi import FastAPI
from langcorn import create_service

app:FastAPI = create_service("langcorn_two_chain:chain", "langcorn_two_chain:chain2")
```

`langcorn_two_chain.py`
```
import os

from langchain import LLMMathChain, OpenAI

os.environ["OPENAI_API_KEY"] = os.environ.get("OPENAI_API_KEY", "sk-********")

llm = OpenAI(temperature=0)
chain = LLMMathChain(llm=llm, verbose=True)
chain2 = LLMMathChain(llm=llm, verbose=True)
```

Then run and load localhost:8000/docs

`uvicorn main:app --host 0.0.0.0 --port 8000`

You'll see:

```
INFO:     127.0.0.1:46330 - "GET /openapi.json HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/home/username/miniconda3/envs/langcorntest/lib/python3.10/site-packages/uvicorn/protocols/http/h11_impl.py", line 428, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/home/username/miniconda3/envs/langcorntest/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
  File "/home/username/miniconda3/envs/langcorntest/lib/python3.10/site-packages/fastapi/applications.py", line 276, in __call__
    await super().__call__(scope, receive, send)
  File "/home/username/miniconda3/envs/langcorntest/lib/python3.10/site-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/username/miniconda3/envs/langcorntest/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/home/username/miniconda3/envs/langcorntest/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/home/username/miniconda3/envs/langcorntest/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/home/username/miniconda3/envs/langcorntest/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/home/username/miniconda3/envs/langcorntest/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "/home/username/miniconda3/envs/langcorntest/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/home/username/miniconda3/envs/langcorntest/lib/python3.10/site-packages/starlette/routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File "/home/username/miniconda3/envs/langcorntest/lib/python3.10/site-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "/home/username/miniconda3/envs/langcorntest/lib/python3.10/site-packages/starlette/routing.py", line 66, in app
    response = await func(request)
  File "/home/username/miniconda3/envs/langcorntest/lib/python3.10/site-packages/fastapi/applications.py", line 231, in openapi
    return JSONResponse(self.openapi())
  File "/home/username/miniconda3/envs/langcorntest/lib/python3.10/site-packages/fastapi/applications.py", line 206, in openapi
    self.openapi_schema = get_openapi(
  File "/home/username/miniconda3/envs/langcorntest/lib/python3.10/site-packages/fastapi/openapi/utils.py", line 423, in get_openapi
    definitions = get_model_definitions(
  File "/home/username/miniconda3/envs/langcorntest/lib/python3.10/site-packages/fastapi/utils.py", line 48, in get_model_definitions
    model_name = model_name_map[model]
KeyError: <class 'abc.LangLangcorn_two_chain'>
```

This error existed in 0.0.9 and before, but it looks like you merged in a fix in 0.0.10.

**Comment by msoedov** [comment]
@kevroy314 I added your example and tests in https://github.com/msoedov/langcorn/commit/43e596b1469a66a71d9b04a2e38244fc1dd247fc. Was not able to reproduce the issue

**Comment by kevroy314** [comment]
The PR I submitted fixes it - you'd need to install langcorn==0.0.9 to see it.
